nodejs/week3/warmup/index.js

```diff
@@ -0,0 +1,65 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+// const { PORT } = process.env || 5000;
+const PORT = 3000;
+
+const app = express();
+
+app.use(bodyParser.json());
+
+app.get('/calculator', (req, res) => {
+	const queryData = Object.values(req.body);
+	res.send(`${calculate(queryData)}`);
+});
+
+// {	"num1": 1,	"num2": 2,	"num3": 4,	"method": "division" }
+function calculate(data) {
+	const method = data[data.length-1];
+	data.pop();
+	if (method === "addition") {
+		return data.reduce((sum, value) => (sum = sum + value), 0);
+	} else if (method === "subtraction") {
+		return data.reduce((sum, value) => (sum = sum - value), 0);
+	} else if (method === "multiplication") {
+		return data.reduce((sum, value) => (sum = sum * value), 1);
+	} else if (method === "division") {
+		return data.reduce((sum, value) => (sum = sum / value), 1);
+	} else {
+		return `Method is not recognized`;
+	}
+}
+
+// calculator/add?firstParam=1&secondParam=2
+app.get('/calculator/add', (req, res) => {
+	const queryData = [].concat.apply([], Object.values(req.query));
```

Comment:
Where do you have this line from? I have seen it in multiple homeworks.

Reply:
Some of us came together for a studygroup at HYF when our class was cancelled before christmas break. We found out that this line works for our homework so I guess we all used it 🙈